src/build/cargo.rs

```diff
@@ -383,6 +385,24 @@ impl Executor for RlsExecutor {
             }
         }
 
+        // Cache executed command for the build plan
+        {
+            let mut cx = self.compilation_cx.lock().unwrap();
+
+            let mut store_cmd = process_builder::process(&rustc_exe);
+            store_cmd.args(&args.iter().map(|s| OsString::from(s)).collect::<Vec<OsString>>());
```

Comment:
You can just use `.map(OsString::from)` here since it's a single parameter function. I believe you should also be able to independently have `collect::<Vec<_>>` too since the map always generates `OsString`s.